### PR TITLE
Expose non-prefixed macros.

### PR DESCRIPTION
**Pull request by floitsch**
Whe DOUBLE_CONVERSION_NON_PREFIXED_MACROS is defined, macros from
utils.h are exposed without the "DOUBLE_CONVERSION_" prefix.

Removing the old macros is still a breaking change, but this should make
it easier to unbreak programs that relied on these macros.

### CODE REVIEW

**Comment by floitsch** [comment]
See https://github.com/google/double-conversion/pull/101#issuecomment-987375886

**Comment by floitsch** [comment]
@ericLemanissier could you verify that this is the expected way of dealing with the change?

File: double-conversion/utils.h [hunk 1]
```diff
@@ -178,6 +210,9 @@ typedef uint16_t uc16;
   TypeName(const TypeName&);                    \
   void operator=(const TypeName&)
 #endif
+#if defined(DOUBLE_CONVERSION_NON_PREFIXED_MACROS) && !defined(DISALLOW_COPY_AND_ASSIGN)
+#define DISALLOW_COPY_AND_ASSIGN DOUBLE_CONVERSION_DISALLOW_COPY_AND_ASSIGN
```
Comment (ericLemanissier) [inline double-conversion/utils.h:214]:
```suggestion
#define DC_DISALLOW_COPY_AND_ASSIGN DOUBLE_CONVERSION_DISALLOW_COPY_AND_ASSIGN
```

File: double-conversion/utils.h [hunk 2]
```diff
@@ -190,6 +225,9 @@ typedef uint16_t uc16;
   TypeName();                                    \
   DOUBLE_CONVERSION_DISALLOW_COPY_AND_ASSIGN(TypeName)
 #endif
+#if defined(DOUBLE_CONVERSION_NON_PREFIXED_MACROS) && !defined(DISALLOW_IMPLICIT_CONSTRUCTORS)
+#define DISALLOW_IMPLICIT_CONSTRUCTORS DOUBLE_CONVERSION_DISALLOW_IMPLICIT_CONSTRUCTORS
```
Comment (ericLemanissier) [inline double-conversion/utils.h:229]:
```suggestion
#define DC_DISALLOW_IMPLICIT_CONSTRUCTORS DOUBLE_CONVERSION_DISALLOW_IMPLICIT_CONSTRUCTORS
```

**Review by ericLemanissier — COMMENTED** [review]
I think this should work

**Comment by floitsch** [comment]
You mean that the `DISALLOW_COPY_AND_ASSIGN` as well as `DISALLOW_IMPLICIT_CONSTRUCTORS` are private enough that they shouldn't be shown with `DOUBLE_CONVERSION_NON_PREFIXED_MACROS` ?

If yes, then I would just not expose them at all, and simply ask users to go for `DOUBLE_CONVERSION_...` if they think it's a good idea.

**Comment by ericLemanissier** [comment]
No,  these 2 macros already had the `DC_` prefix, I'm just restoring the existing macros

**Comment by floitsch** [comment]
Ah right.
Thanks for catching that.

File: double-conversion/utils.h [hunk 3]
```diff
@@ -190,6 +225,9 @@ typedef uint16_t uc16;
   TypeName();                                    \
   DOUBLE_CONVERSION_DISALLOW_COPY_AND_ASSIGN(TypeName)
 #endif
+#if defined(DOUBLE_CONVERSION_NON_PREFIXED_MACROS) && !defined(DISALLOW_IMPLICIT_CONSTRUCTORS)
```
Comment (floitsch) [inline double-conversion/utils.h:228]:
```suggestion
#if defined(DOUBLE_CONVERSION_NON_PREFIXED_MACROS) && !defined(DC_DISALLOW_IMPLICIT_CONSTRUCTORS)
```

File: double-conversion/utils.h [hunk 4]
```diff
@@ -178,6 +210,9 @@ typedef uint16_t uc16;
   TypeName(const TypeName&);                    \
   void operator=(const TypeName&)
 #endif
+#if defined(DOUBLE_CONVERSION_NON_PREFIXED_MACROS) && !defined(DISALLOW_COPY_AND_ASSIGN)
```
Comment (floitsch) [inline double-conversion/utils.h:213]:
```suggestion
#if defined(DOUBLE_CONVERSION_NON_PREFIXED_MACROS) && !defined(DC_DISALLOW_COPY_AND_ASSIGN)
```

**Review by floitsch — COMMENTED** [review]
Fix two refs.

**Pull request merged**